src/plugins/data/common/search/search_source/legacy/fetch_soon.ts

```diff
@@ -83,20 +83,27 @@ async function delayedFetch(
   if (ms === 0) {
     return callClient([request], [options], fetchHandlers)[0];
   }
+  const { dataSourceId, ...restRequest } = request;
```

Comment:
Not sure if it is straightforward to add unit test for this function but it will be great that we have one, the logic is kind of complex.